pkg/cluster/cluster_config.go

```diff
@@ -287,3 +288,56 @@ func IsDefaultAuthMethod(ctx context.Context, cli client.Client) (bool, error) {
 	// we only create userGroups for "IntegratedOAuth" or "" and leave other or new supported type value in the future
 	return authenticationobj.Spec.Type == configv1.AuthenticationTypeIntegratedOAuth || authenticationobj.Spec.Type == "", nil
 }
+
+func GetReadyWorkerNodes(ctx context.Context, k8sclient client.Client) ([]corev1.Node, error) {
+	nodeList := &corev1.NodeList{}
+
+	labelSelector := client.MatchingLabels{
+		labels.WorkerNode: "",
+	}
+
+	if err := k8sclient.List(ctx, nodeList, labelSelector); err != nil {
+		return nil, fmt.Errorf("failed to list worker nodes: %w", err)
+	}
+
+	var readyWorkerNodes []corev1.Node
+	for _, node := range nodeList.Items {
+		if node.Status.Conditions != nil {
```

Comment:
There are different opinions, but I personally prefer `if node.Status.Conditions == nil { continue }` in such places.

Reply:
Fixed in latest commit